Docker/UsefulCommands.md

```diff
@@ -0,0 +1,141 @@
+# Useful Docker Commands
+Here follows a list of useful Docker commands with useful flags for each
+command.
+
+# Table of Contents
+0. [`docker build`](#docker-build)
+0. [`docker exec`](#docker-exec)
+0. [`docker images`](#docker-images)
+0. [`docker inspect`](#docker-inspect)
+0. [`docker logs`](#docker-logs)
+0. [`docker ps`](#docker-ps)
+0. [`docker rmi`](#docker-rmi)
+0. [`docker run`](#docker-run)
+0. [Learn More](#learn-more)
+
+
+## [`docker build`](https://docs.docker.com/engine/reference/commandline/build/)
+Build an image from a Dockerfile.
+
+```sh
+docker build [DOCKERFILE PATH]
+```
+
+### Example
+Build an image tagged `my-org:my-image` where the Dockerfile can be found at
+`/tmp/Dockerfile`.
+
+```sh
+docker build -t my-org:my-image -f /tmp/Dockerfile
+```
+
+### Useful flags
+- `--file -f` Path where to find the Dockerfile
+- `--force-rm` Always remove intermediate containers
+- `--no-cache` Do not use cache when building the image
+- `--rm` Remove intermediate containers after a successful build (this is
+`true`) by default
+- `--tag -t` Name and optionally a tag in the ‘name:tag’ format
+
+
+## [`docker exec`](https://docs.docker.com/engine/reference/commandline/exec/)
+Execute a command inside a **running** container.
+
+```sh
+docker exec [CONTAINER ID]
+```
+
+### Example
+```sh
+docker exec [CONTAINER ID] touch /tmp/exec_works
+```
+
+### Useful flags
+- `--detach -d` Detached mode: run command in the background
+- `-it` This will not make the container you started shut down immediately, as
+it will create a pseudo-TTY session (`-t`) and keep STDIN open (`-i`)
+
+
+## [`docker images`](https://docs.docker.com/engine/reference/commandline/images/)
+List all downloaded/created images.
+
+```sh
+docker images
+```
+
+### Useful flags
+- `-q` Only show numeric IDs
+
+
+## [`docker inspect`](https://docs.docker.com/engine/reference/commandline/inspect)
+Shows all the info of a container.
+
+```sh
+docker inspect [CONTAINER ID]
+```
+
+
+## [`docker logs`](https://docs.docker.com/engine/reference/commandline/logs/)
+Gets logs from container.
+
+```sh
+docker logs [CONTAINER ID]
+```
+
+### Useful flags
+- `--details` Log extra details
+- `--follow -f` Follow log output
```

Comment:
Maybe add additional info what does "follow" mean? (I know all the descriptions are taken from `docker help [command]`)

---

For example this is taken from `man tail`

> The `-f` option causes tail to not stop when end of file is reached, but rather to wait for additional data to be appended to the input.  The `-f` option is ignored if the standard input is a pipe, but not if it is a FIFO.